internal/core/items/text.rs

```diff
@@ -579,6 +573,9 @@ impl Item for TextInput {
                         window_adapter.input_method_request(InputMethodRequest::Disable {});
                     }
                 }
+                self.preedit_text.set(Default::default());
+                self.preedit_selection_start.set(0);
+                self.preedit_selection_end.set(0);
```

Comment:
Why?  if the edit became read_only or anything, should we not also reset the preedit?

Reply:
I think the two need to be in sync. If for example the text input thinks that there's no pre-edit but the input method still thinks there is, you end up with chaos (non-sensible suggestions from the input method for example). This happens exactly when the text input transitions from read-write to read-only: We clear the pre-edit fields on our edit, but we do _not_ inform the input method that it needs to be disabled.

This is slightly orthogonal, which is why I made the above as a suggested change and approved it (as opposed to flagging it as "requires changes"). Orthogonal because the current code is broken: We can't detect when that transition happens. At this point here in the code it's too late, the `read_only` property will already return true and we fail to send `InputMethodRequest::Disable`. But conceptually sending this requires should IMO happen in the same "code block" as us clearing the local pre-edit state.